### ISSUE TITLE
 /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found

### ISSUE THREAD

**Issue by sophia-guo** [adoptium/infrastructure#3740]
/lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found on https://ci.adoptium.net/computer/test%2Ddocker%2Dubuntu2004%2Dx64%2D4/

https://ci.adoptium.net/view/Test_grinder/job/Grinder_Simple/391/
Same job no issues on https://ci.adoptium.net/computer/test-docker-ubuntu2204-x64-6/
https://ci.adoptium.net/view/Test_grinder/job/Grinder_Simple/392/

**Comment by sxa** [comment]
It looks like you're pulling a build from a third party which has been built targeting a newer version of glibc than is shipped with Ubuntu 20.04. 

This seems to be "working as designed" and that build is not a Temurin build and is not compatible with that distribution so I'm not sure what you are expecting us to do here.

**Comment by sophia-guo** [comment]
It seems Temurin build has the same issue - jdk from https://github.com/adoptium/temurin24-binaries/releases/tag/jdk-24%2B14-ea-beta

https://ci.adoptium.net/view/Test_grinder/job/Grinder/11142/console

Does this mean for jdk24+ GLIBC_2.14 is a must? @andrew-m-leonard

**Comment by sxa** [comment]
Temurin 21+ on Linux/x64 is built on CentOS/RHEL7 and will not run on machines with a glibc version earlier than the ones on those distributions. This is why the pipelines for those versions have
```
"ADDITIONAL_TEST_LABEL": "!(centos6||rhel6)",`
```
https://github.com/adoptium/ci-jenkins-pipelines/pull/1005 was the PR which introduced this (which you approved ;-) )

**Comment by sxa** [comment]
> Does this mean for jdk24+ GLIBC_2.14 is a must? @andrew-m-leonard

See footnote 1 in the table on on [the supported platforms page](https://adoptium.net/en-GB/supported-platforms/) which tells you that it requires 2.17

**Comment by sophia-guo** [comment]
yeah, ADDITIONAL_TEST_LABEL https://github.com/adoptium/ci-jenkins-pipelines/pull/1005 does solve the issue if tests builds are triggered by build pipeline. Though it means for grinder, users have to figure out this themselves. Could always forgot...

**Comment by smlambert** [comment]
Re-opening this one to ensure we apply an appropriate test label (following an agreed upon labelling schema so it interoperates across all Jenkins servers) in the test pipeline code (rather than the build pipeline code), to avoid continuously hitting this when running Grinders.  

https://ci.adoptium.net/view/Test_grinder/job/Grinder/11386/console
